### DISCUSSION TITLE
maps.Mesh warning when passing it a string ?

### DISCUSSION THREAD

**Discussion by functionstackx** [jax-ml/jax#13201] [Ideas] [answered]
i think it would be good to warning if a user passed into a string into maps.Mesh

```python3
mesh = maps.Mesh(devices, ("dp")) # forgot the comma
print(mesh.axis_names)
```
results in
```
('d','p')
```

due to the tuple function at https://github.com/google/jax/blob/main/jax/interpreters/pxla.py#L2324

I can see it as a very common mistake that users run into and spend 15 mins debugging  (i fell victim to forgetting a comma)

Proper Code without mistake
```python3
mesh = maps.Mesh(devices, ("dp",)) 
print(mesh.axis_names)
```
results in
```
('dp')
```

**Comment by sharadmv** [comment, marked as answer]
I think we can just convert a string input to a singleton tuple (see https://github.com/google/jax/pull/13213)
